src/services/notifications/index.ts

```diff
@@ -166,19 +173,26 @@ export const NotificationsService = (logger: Logger): INotificationsService => {
       publish(senderWalletId, NotificationType.IntraLedgerPayment)
       publish(recipientWalletId, NotificationType.IntraLedgerReceipt)
 
-      // work around to move forward before re-wrighting the whole notifications module
-      const payerUser = await User.findOne({ walletId: senderWalletId })
+      const senderAccount = await AccountsRepository().findByWalletId(senderWalletId)
+      if (senderAccount instanceof Error) return senderAccount
+
+      const senderUser = await UsersRepository().findById(senderAccount.ownerId)
+      if (senderUser instanceof Error) return senderUser
 
       // Do not await this call for quicker processing
       transactionNotification({
         type: NotificationType.IntraLedgerPayment,
-        user: payerUser,
+        user: senderUser,
         logger,
         amount,
         usdPerSat,
       })
 
-      const recipientUser = await User.findOne({ walletId: recipientWalletId })
+      const recipientAccount = await AccountsRepository().findByWalletId(senderWalletId)
```

Comment:
please use `recipientWalletId` instead of `senderWalletId`

Reply:
good catch.